src/Shared/LayoutRenderers/AspNetRequestPostedBodyLayoutRenderer.cs

```diff
@@ -17,14 +18,44 @@ namespace NLog.Web.LayoutRenderers
     [LayoutRenderer("aspnet-request-posted-body")]
     public class AspNetRequestPostedBodyLayoutRenderer : AspNetLayoutRendererBase
     {
+        internal static bool MiddlewareInstalled;
+
         /// <summary>
         /// The object for the key in HttpContext.Items for the POST request body
         /// </summary>
         internal static readonly object NLogPostedRequestBodyKey = new object();
 
+        /// <inheritdoc />
+        protected override void InitializeLayoutRenderer()
+        {
+            base.InitializeLayoutRenderer();
+        }
+
+        private static bool LogEventProcessed;
+        private static readonly object LogEventProcessedLock = new object();
+
         /// <inheritdoc/>
         protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
         {
+            if (!LogEventProcessed)
+            {
+                lock (LogEventProcessedLock)
```

Comment:
Though I like the idea of only printing the alert exactly once, then I think using a monitor-lock-object is a little too heavy. Could it just be (Removing `LogEventProcessedLock`):
```c#
if (!LogEventProcessed)
{
   LogEventProcessed = true;
   if (!MiddlewareInstalled)
   {
       ...
   }
}
```